test/ProjectService.Business.UnitTests/Commands/CreateTaskCommandTests.cs

```diff
@@ -0,0 +1,151 @@
+                Number = 2,
+                PriorityId = Guid.NewGuid(),
+                StatusId = Guid.NewGuid(),
+                TypeId = Guid.NewGuid()
+            };
+
+            _dbTask = new DbTask
+            {
+                Id = Guid.NewGuid(),
+                Name = "Create Smth",
+                Description = "Create smth in somewhere",
+                PlannedMinutes = 30,
+                AssignedTo = _newRequest.AssignedTo,
+                AuthorId = _newRequest.AuthorId,
+                Deadline = _newRequest.Deadline,
+                ProjectId = _newRequest.ProjectId,
+                CreatedAt = _newRequest.CreatedAt,
+                ParentTaskId = _newRequest.ParentTaskId,
+                Number = 2,
+                Priority = new DbTaskProperty()
+                {
+                    Id = _newRequest.PriorityId
+                },
+                Status = new DbTaskProperty()
+                {
+                    Id = _newRequest.StatusId
+                },
+                Type = new DbTaskProperty()
+                {
+                    Id = _newRequest.TypeId
+                }
+            };
+
+            _response = new OperationResultResponse<Guid>
+            {
+                Body = _dbTask.Id
+            };
+        }
+
+        [SetUp]
+        public void SetUp()
+        {
+            _mocker.GetMock<ICreateTaskValidator>().Reset();
+            _mocker.GetMock<IAccessValidator>().Reset();
+            _mocker.GetMock<IDbTaskMapper>().Reset();
+            _mocker.GetMock<ITaskRepository>().Reset();
+        }
+
+        [Test]
+        public void ShouldThrowExceptionWhenCreatingNewTaskWithIncorrectTasktData()
+        {
+            _mocker
+                .Setup<ICreateTaskValidator, bool>(x => x.Validate(It.IsAny<IValidationContext>()).IsValid)
+                .Returns(false);
+
+            Assert.Throws<ValidationException>(() => _command.Execute(_newRequest));
+            _mocker.Verify<ICreateTaskValidator, bool>(x => x.Validate(It.IsAny<IValidationContext>()).IsValid, Times.Once);
+        }
+
+        [Test]
+        public void ShouldReturnResponseWhenCreatingNewTaskAndUserHasRights()
+        {
+            _mocker
+                .Setup<ICreateTaskValidator, bool>(x => x.Validate(It.IsAny<IValidationContext>()).IsValid)
+                .Returns(true);
+
+            _mocker
+                .Setup<IDbTaskMapper, DbTask>(x => x.Map(_newRequest, _authorId))
+                .Returns(_dbTask);
+
+            _mocker
+                .Setup<ITaskRepository, Guid>(x => x.CreateTask(_dbTask))
+                .Returns(_response.Body);
+
+            SerializerAssert.AreEqual(_response, _command.Execute(_newRequest));
+        }
+
+        [Test]
+        public void ShouldReturnResponseWhenCreatingNewTaskAndUserIsAdmin()
+        {
+            _mocker
+               .Setup<IAccessValidator, bool>(x => x.IsAdmin())
+               .Returns(true);
+
```

Comment:
In CreateTaskCommand not exist checks of AccessValidator, why are you mock this? Maybe should add AccessValidator checks  to the command?

Reply:
fixed